samples/azure/azure-service-fabric-routing/sample.md

```diff
@@ -0,0 +1,202 @@
+
+
+## Routing
+
+The default NServiceBus routing approach cannot be used as-is with Service Fabric stateful services. Stateful services assume business data partitioning. A message must be delivered to a specific replica (instance) of a stateful that can handle the message data. E.g. for each `PlaceVote` message associated with a cast vote, the message should be routed to the partition associated with the preferred candidate, "John" or "Abby". Similar to that, to count the votes per zip code the `TrackZipCode` message needs to delivered to the partition that is responsible for the range to which the zip code belongs.
+
+Example:
+
+- Vote for John; cast in 88701; should result in a `PlaceVote` message routed to named partition "John", followed by a `TrackZipCode` message routed to range partition (66000 -> 98999).
+- Vote for Abby; cast in 36458; should result in a `PlaceVote` message routed to named partition "Abby", followed by a `TrackZipCode` message routed to range partition (33000 -> 65999).
+- Vote for John; cast in 12789; should result in a `PlaceVote` message routed to named partition "John", followed by a `TrackZipCode` message routed to range partition (0 -> 32999).
+
+Partition-aware behavior is achieved by combining sender-side distribution (a built-in NServiceBus feature) with a few modifications to the message processing pipeline.
+
+The remainder of this document will focus on the different techniques that can be used to configure these distribution strategies, either manually or automatically, to achieve full partition aware routing. 
+
+
+## Partitioned Endpoint Configuration
+
+Endpoint instances hosted with stateful service replicas need to be uniquely addressable by the partition key associated with each replica. Possible keys are defined by the `NamedPartition` and `Int64RangePartition` info obtained from Service Fabric. The unique partition keys are used for `SendLocal` and `ReplyTo` operations to route messages to the correct partitions.
+
+Partitioned endpoints require specifying partitioning configuration by calling an extension method on `EndpointConfiguration`:
+
+snippet: ApplyPartitionConfigurationToEndpoint-ZipCodeVoteCount
+
+### Local sends
+
+All local sends are handled by the `PartitionAwareDistributionStrategy`. For each locally sent message it sets the `partition-key` header value to the local partition key and routes the message to the queue associated with the local partition.
+
+### Replies
+
+When replying, an endpoint routes the reply message to the endpoint that initiated the conversation. It's the responsibility of the requestor to properly set the reply-to header before sending out the request. For a partitioned endpoint this implies that it sets the reply to header to its instance specific queue instead of the shared queue. This functionality is covered by the `HardcodeReplyToAddressToLogicalAddress` behavior.
```

Comment:
The intention here was that local sends and reply-to are `partition stick`. 

The partition id is not the discriminator. In general there is no single partition id concept in SF e.g. there range partitions that cover a range of values and not a single one. It is true though that instance discriminator is fully defined by partition configuration.